library/core/src/mem/maybe_uninit.rs

```diff
@@ -839,7 +839,9 @@ impl<T> MaybeUninit<T> {
         // * MaybeUnint does not drop, so there are no double-frees
         // And thus the conversion is safe
         unsafe {
-            intrinsics::assert_inhabited::<T>();
```

Comment:
This could assert that `[T; N]` is inhabited without extra conditional.

Reply:
Good idea. I didn't realize that.